mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -440,12 +440,9 @@ final class OnlineLDAOptimizer extends LDAOptimizer {
       val stat = BDM.zeros[Double](k, vocabSize)
       var gammaPart = List[BDV[Double]]()
       nonEmptyDocs.foreach { case (_, termCounts: Vector) =>
-        val ids: List[Int] = termCounts match {
-          case v: DenseVector => (0 until v.size).toList
-          case v: SparseVector => v.indices.toList
-        }
+        val ids: List[Int] = LDAUtils.vectorAsList(termCounts)
```

Comment:
I see what you're optimizing here, but the cost is making all the other callers compute this. Is it enough of a performance bottleneck to bother? Is it at all sensible to simply return `ids` from the existing method call instead?